pages/api/getWebLogin.js

```diff
@@ -0,0 +1,24 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+
+import getConfig from "next/config";
+
+const { serverRuntimeConfig } = getConfig();
+
+export default function getWebLogin(req, res) {
+  fetch(serverRuntimeConfig.HUMAN_ID_GET_WEB_LOGIN_URL, {
+    method: "POST",
+    headers: {
+      "client-id": serverRuntimeConfig.HUMAN_ID_CLIENT_ID,
+      "client-secret": serverRuntimeConfig.HUMAN_ID_CLIENT_SECRET,
+      "Content-Type": "application/json",
+    },
+  })
+    .then(async (result) => {
+      const data = await result.json();
+      res.status(200).json(data);
+    })
+    .catch((error) => {
+      console.log(error);
+      res.status(500).json(error);
+    });
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getWebLogin` function is directly exposing server-side secrets (`HUMAN_ID_CLIENT_ID`, `HUMAN_ID_CLIENT_SECRET`) to the client. This is a security risk as these secrets can be intercepted and misused. Instead, you should create a server-side API that interacts with the `HUMAN_ID_GET_WEB_LOGIN_URL` and returns only the necessary data to the client.

Also, error handling in this function could be improved. Currently, if an error occurs during the fetch operation, the error object is sent directly to the client (line 22). This could potentially expose sensitive information. It would be better to send a generic error message to the client and log the specific error details server-side for debugging.

Here's how you might refactor this code:

```diff
- export default function getWebLogin(req, res) {
-   fetch(serverRuntimeConfig.HUMAN_ID_GET_WEB_LOGIN_URL, {
-     method: "POST",
-     headers: {
-       "client-id": serverRuntimeConfig.HUMAN_ID_CLIENT_ID,
-       "client-secret": serverRuntimeConfig.HUMAN_ID_CLIENT_SECRET,
-       "Content-Type": "application/json",
-     },
-   })
-     .then(async (result) => {
-       const data = await result.json();
-       res.status(200).json(data);
-     })
-     .catch((error) => {
-       console.log(error);
-       res.status(500).json(error);
-     });
- }

+ import axios from 'axios';
+
+ export default async function getWebLogin(req, res) {
+   try {
+     const response = await axios.post(serverRuntimeConfig.HUMAN_ID_GET_WEB_LOGIN_URL, {}, {
+       headers: {
+         "client-id": serverRuntimeConfig.HUMAN_ID_CLIENT_ID,
+         "client-secret": serverRuntimeConfig.HUMAN_ID_CLIENT_SECRET,
+         "Content-Type": "application/json",
+       },
+     });
+
+     res.status(200).json(response.data);
+   } catch (error) {
+     console.error(error);
+     res.status(500).json({ message: 'An error occurred while processing your request.' });
+   }
+ }
```
This refactored code uses `axios` for making HTTP requests, which simplifies the code by removing the need to convert the response to JSON manually. It also improves error handling by logging the error server-side and sending a generic error message to the client.


<!-- This is an auto-generated comment by OSS CodeRabbit -->